go/oasis-test-runner/scenario/e2e/change_reward_schedule.go

```diff
@@ -0,0 +1,451 @@
+	err = sc.Net.Controller().Consensus.SubmitTx(ctx, sigTx)
+	if err != nil {
+		return nil, fmt.Errorf("failed submitting proposal transaction: %w", err)
+	}
+
+	// Ensure proposal created.
+	aps, err := sc.Net.Controller().Governance.ActiveProposals(ctx, consensus.HeightLatest)
+	if err != nil {
+		return nil, fmt.Errorf("failed querying active proposals: %w", err)
+	}
+	var proposal *api.Proposal
+	for _, p := range aps {
+		if p.Content.Equals(content) {
+			proposal = p
+			break
+		}
+	}
+	if proposal == nil {
+		return nil, fmt.Errorf("submitted proposal %v not found", content)
+	}
+
+	// Vote for the proposal.
+	vote := api.ProposalVote{
+		ID:   proposal.ID,
+		Vote: api.VoteYes,
+	}
+	tx = api.NewCastVoteTx(sc.entityNonce, &transaction.Fee{Gas: 2000}, &vote)
+	sc.entityNonce++
+	sigTx, err = transaction.Sign(sc.entity.Signer(), tx)
+	if err != nil {
+		return nil, fmt.Errorf("failed signing cast vote transaction: %w", err)
+	}
+	sc.Logger.Info("submitting vote for proposal", "proposal", proposal, "vote", vote)
+	err = sc.Net.Controller().Consensus.SubmitTx(ctx, sigTx)
+	if err != nil {
+		return nil, fmt.Errorf("failed submitting cast vote transaction: %w", err)
+	}
+
+	// Ensure vote was cast.
+	votes, err := sc.Net.Controller().Governance.Votes(ctx,
+		&api.ProposalQuery{
+			Height:     consensus.HeightLatest,
+			ProposalID: aps[0].ID,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed queying votes: %w", err)
+	}
+	if l := len(votes); l != 1 {
+		return nil, fmt.Errorf("expected one vote, got: %v", l)
+	}
+	if vote := votes[0].Vote; vote != api.VoteYes {
+		return nil, fmt.Errorf("expected vote Yes, got: %s", string(vote))
+	}
+
+	// Transition to the epoch when proposal finalizes.
+	for ep := sc.currentEpoch + 1; ep < aps[0].ClosesAt+1; ep++ {
+		sc.Logger.Info("transitioning to epoch", "epoch", ep)
+		if err = sc.nextEpoch(); err != nil {
+			return nil, err
+		}
+	}
+
+	p, err := sc.Net.Controller().Governance.Proposal(ctx,
+		&api.ProposalQuery{
+			Height:     consensus.HeightLatest,
+			ProposalID: proposal.ID,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query proposal: %w", err)
+	}
+	sc.Logger.Info("got proposal",
+		"state", p.State.String(),
+		"results", p.Results,
+		"len", len(p.Results),
+		"invalid", p.InvalidVotes,
+	)
+	// Ensure proposal finalized.
+	if p.State == api.StateActive || p.State == api.StateFailed {
+		return nil, fmt.Errorf("expected finalized proposal, proposal state: %v", p.State)
+	}
+
+	return p, nil
+}
+
+func (sc *changeRewardScheduleImpl) getAccount(owner staking.Address) (*staking.Account, error) {
+	a, err := sc.Net.Controller().Staking.Account(sc.ctx,
+		&staking.OwnerQuery{
+			Height: consensus.HeightLatest,
+			Owner:  owner,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed querying account: %w", err)
+	}
+
+	return a, nil
+}
+
+func (sc *changeRewardScheduleImpl) getEscrowBalance(owner staking.Address) (*quantity.Quantity, error) {
```

Comment:
This is not an accessor, but actually needs to do work to get the balance, so that's why `get` is part of the name :P
It also follows the naming style of other method names, e.g. `ensureProposalFinalized`, etc.
I'd prefer to keep the verb in the name (maybe replace it with `fetch`), because this balance is not stored locally, but needs to be fetched via the client.

Reply:
You are right. A name starting with `get` is usually confusing as reminds you on Java and you think that this is an accessor. I agree that `fetch` is a better alternative.